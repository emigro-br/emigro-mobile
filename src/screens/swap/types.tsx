
export enum SwapType {
   SELL,
   BUY
}
