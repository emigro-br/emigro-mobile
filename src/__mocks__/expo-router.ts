import { ExpoRouter } from 'expo-router/types/expo-router';

const routerMock = {
  back: jest.fn(),
  canGoBack: jest.fn(),
  push: jest.fn(),
  navigate: jest.fn(),
  replace: jest.fn(),
  dismiss: jest.fn(),
  dismissAll: jest.fn(),
  canDismiss: jest.fn(),
  setParams: jest.fn(),
};

export const useRouter = (): ExpoRouter.Router => routerMock;

export const useLocalSearchParams = jest.fn().mockImplementation(() => ({}));

export const useFocusEffect = jest.fn();

const stackMock = {
  Screen: jest.fn(),
  Navigator: jest.fn(),
};

export const Stack = stackMock;
