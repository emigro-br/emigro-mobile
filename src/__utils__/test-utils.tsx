import { JSX, JSXElementConstructor, ReactElement } from 'react';

import { RenderOptions, render } from '@testing-library/react-native';

import { ThemeProvider } from './ThemeProvider';

type Props = {
  children: JSX.Element;
};

const AllTheProviders = ({ children }: Props) => {
  return <ThemeProvider theme="light">{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement<unknown, string | JSXElementConstructor<any>>,
  options?: RenderOptions | undefined,
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
