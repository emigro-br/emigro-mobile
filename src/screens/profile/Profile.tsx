import { View } from 'react-native';

import { NativeStackNavigationProp } from '@react-navigation/native-stack';

import {
  Avatar,
  AvatarFallbackText,
  Box,
  Button,
  ButtonIcon,
  ButtonText,
  Center,
  CopyIcon,
  Divider,
  Heading,
  Spinner,
  Text,
  Toast,
  ToastDescription,
  VStack,
  useToast,
} from '@gluestack-ui/themed';
import * as Application from 'expo-application';
import * as Clipboard from 'expo-clipboard';
import { observer } from 'mobx-react-lite';

import { ProfileStackParamList } from '@navigation/ProfileStack';

import { sessionStore } from '@stores/SessionStore';

import { maskWallet } from '@utils/masks';

type Props = {
  navigation: NativeStackNavigationProp<ProfileStackParamList, 'Profile'>;
};

const Profile = observer(({ navigation }: Props) => {
  const toast = useToast();

  const handleLogout = async () => {
    await sessionStore.clear();
  };

  const publicKey = sessionStore.publicKey;

  const copyToClipboard = async () => {
    if (publicKey) {
      await Clipboard.setStringAsync(publicKey);
      toast.show({
        render: ({ id }) => (
          <Toast nativeID={`toast${id}`} action="info" variant="solid">
            <ToastDescription>Copied to clipboard</ToastDescription>
          </Toast>
        ),
      });
    }
  };

  const profileInfo = sessionStore.profile;
  if (!profileInfo) {
    return (
      <Center flex={1} bg="$backgroundLight0">
        <Spinner size="large" testID="loading" />
      </Center>
    );
  }

  const fullName = `${profileInfo.given_name} ${profileInfo.family_name}`;

  return (
    <Box flex={1} bg="$white" justifyContent="space-between">
      <VStack p="$4" space="lg">
        <Center>
          <Avatar bgColor="$primary300" size="xl" borderRadius="$full">
            <AvatarFallbackText>{fullName}</AvatarFallbackText>
          </Avatar>
          <Heading py="$2">{fullName}</Heading>
          {publicKey && (
            <Button size="md" variant="link" action="primary" onPress={copyToClipboard}>
              <ButtonText>{maskWallet(publicKey)}</ButtonText>
              <ButtonIcon as={CopyIcon} ml="$2" />
            </Button>
          )}
        </Center>

        <VStack space="xl">
          <View>
            <Text size="sm" color="$textLight500">
              Full Name
            </Text>
            <Text>{fullName}</Text>
          </View>

          <Divider />

          <View>
            <Text size="sm" color="$textLight500">
              Email address
            </Text>
            <Text>{profileInfo.email}</Text>
          </View>

          <Divider />

          <View>
            <Text size="sm" color="$textLight500">
              Address
            </Text>
            <Text>{profileInfo.address}</Text>
          </View>

          <Divider />

          <Button onPress={() => navigation.push('DeleteAccount')} variant="link" size="sm" alignSelf="flex-start">
            <ButtonText>Delete account</ButtonText>
          </Button>
        </VStack>
      </VStack>
      <Box alignItems="center" py="$4">
        <Button onPress={handleLogout} variant="link" size="lg">
          <ButtonText>Logout</ButtonText>
        </Button>
        <Text size="sm">
          version {Application.nativeApplicationVersion} ({Application.nativeBuildVersion})
        </Text>
      </Box>
    </Box>
  );
});

export default Profile;
