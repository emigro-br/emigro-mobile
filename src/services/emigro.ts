import { IBalance } from '@/types/IBalance';
import { IPaymentResponse } from '@/types/IPaymentResponse';
import { IQuote } from '@/types/IQuote';
import { IQuoteRequest } from '@/types/IQuoteRequest';
import { ITransaction } from '@/types/ITransaction';
import { ITransactionRequest } from '@/types/ITransactionRequest';
import { IUserProfile } from '@/types/IUserProfile';
import { refresh as refreshSession } from '@/services/auth';
import { getSession, saveSession } from '@/storage/helpers';

import { GET_USER_BALANCE_ERROR, QUOTE_NOT_AVAILABLE_ERROR, TRANSACTION_ERROR_MESSAGE } from '@constants/errorMessages';

const backendUrl = process.env.EXPO_PUBLIC_BACKEND_URL;


export class NotAuhtorized extends Error {
  constructor() {
    super('Not authorized');
  }
}

export const fetchWithTokenCheck = async (url: string, options: RequestInit): Promise<Response> => {

  let session = await getSession();
  if (!session) {
    throw new NotAuhtorized();
  }

  const { tokenExpirationDate } = session;
  const isTokenExpired = tokenExpirationDate < new Date();

  if (isTokenExpired) {
    console.debug('Token expired, refreshing...');
    const newSession = await refreshSession(session);
    if (newSession) {
      saveSession(newSession);
      session = newSession;
    } else {
      throw new Error('Could not refresh session');
    }
  }

  const { accessToken } = session;
  options.headers = {
    ...options.headers,
    Authorization: `Bearer ${accessToken}`,
  };

  return fetch(url, options);
}

export const getTransactions = async (): Promise<ITransaction[]> => {
  const transactionsUrl = `${backendUrl}/transaction/all`;

  try {
    const response = await fetchWithTokenCheck(transactionsUrl, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
      },
    });
    const { transactions } = await response.json();
    return transactions;
  } catch (error) {
    console.error(error);
    throw new Error();
  }
};

export const getUserBalance = async (): Promise<IBalance[]> => {
  const url = `${backendUrl}/user`;
  try {
    const response = await fetchWithTokenCheck(url, {
      method: 'GET',
    });
    const json = await response.json();

    if (json.statusCode === 401) {
      throw new NotAuhtorized();
    }

    const { balances } = json;
    return balances;
  } catch (error) {
    console.error(error, GET_USER_BALANCE_ERROR);
    throw new Error();
  }
};

export const handleQuote = async (body: IQuoteRequest): Promise<IQuote> => {
  const url = `${backendUrl}/quote`;

  try {
    const response = await fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(body),
    });
    const { quote } = await response.json();
    return quote;
  } catch (error) {
    console.error(error, QUOTE_NOT_AVAILABLE_ERROR);
    throw new Error();
  }
};

export const sendTransaction = async (transactionRequest: ITransactionRequest): Promise<IPaymentResponse> => {
  const url = `${backendUrl}/transaction`;
  try {
    const response = await fetchWithTokenCheck(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(transactionRequest),
    });
    return await response.json();
  } catch (error) {
    console.error(error, TRANSACTION_ERROR_MESSAGE);
    throw new Error();
  }
};

export const getUserPublicKey = async (): Promise<string> => {
  const url = `${backendUrl}/user`;

  try {
    const request = await fetchWithTokenCheck(url, {
      method: 'GET',
    });
    const { publicKey } = await request.json();
    return publicKey;
  } catch (error) {
    console.error(error, GET_USER_BALANCE_ERROR);
    throw error;
  }
};

export const getUserProfile = async (): Promise<IUserProfile> => {
  const url = `${backendUrl}/user/profile`;
  try {
    const session = await getSession();
    const response = await fetchWithTokenCheck(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(session),
    });
    const json = await response.json();
    if (json.statusCode === 401) {
      throw new NotAuhtorized();
    }
    return json
  } catch (error) {
    console.error(error);
    throw error;
  }
};
