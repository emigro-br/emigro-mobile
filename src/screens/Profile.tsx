import { styled } from 'nativewind';
import { useNavigation } from '@react-navigation/native';
import React, { useEffect, useState } from 'react';
import { ActivityIndicator, Image, ScrollView, Text, View } from 'react-native';
import * as Application from 'expo-application';
import { clearSession } from '@/storage/helpers';

import { getUserProfile } from '@/services/emigro';

import profileLogo from '@assets/images/profile-icon.png';

const StyledScrollView = styled(ScrollView);
const StyledView = styled(View);
const StyledText = styled(Text);
const StyledImage = styled(Image);

const Profile = () => {
  const navigation = useNavigation();
  const [userInformation, setUserInformation] = useState<any>(null);

  const handleLogout = async () => {
    await clearSession();
    navigation.navigate('Welcome' as never);
  };


  useEffect(() => {
    const fetchUserInformation = async () => {
      try {
        const userProfile = await getUserProfile();
        if (userProfile) {
          setUserInformation(userProfile);
        }
      } catch (error) {
        console.error(error);
      }
    };
    fetchUserInformation();
  }, []);

  if (!userInformation) {
    return (
      <StyledView className="flex items-center justify-center h-full">
        <ActivityIndicator size="large" />
      </StyledView>
    );
  }

  return (
    <StyledScrollView>
      <StyledView className="flex items-center pt-8 px-4 bg-white h-full">
        <StyledView className="mb-8">
          <StyledImage source={profileLogo} className="h-32 w-32" />
        </StyledView>

        <StyledView className="flex gap-2 w-full">
          <StyledText className="text-lightGray">Full name</StyledText>
          <StyledView className="flex flex-row items-center">
            <StyledText className="text-lg">
              {userInformation.given_name} {userInformation.family_name}
            </StyledText>
          </StyledView>

          <StyledView className="border-b-[1px] border-slate-200 w-full pt-4 mb-4" />

          <StyledText className="text-lightGray">Email address</StyledText>
          <StyledView className="flex flex-row items-center">
            <StyledText className="text-lg">{userInformation.email}</StyledText>
          </StyledView>

          <StyledView className="border-b-[1px] border-slate-200 w-full pt-4 mb-4" />

          <StyledText className="text-lightGray">Address</StyledText>
          <StyledView className="flex flex-row items-center">
            <StyledText className="text-lg">{userInformation.address}</StyledText>
          </StyledView>

        </StyledView>

        
        <StyledView className="flex items-center pt-8 mb-2">
          <StyledText className="text-red text-lg py-2" onPress={handleLogout}>
            Log out
          </StyledText>
          <StyledText className="text-lightGray">ver. {Application.nativeApplicationVersion} ({Application.nativeBuildVersion})</StyledText>
        </StyledView>

      </StyledView>
    </StyledScrollView>
  );
};

export default Profile;
