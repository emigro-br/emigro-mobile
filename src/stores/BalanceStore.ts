import { action, makeAutoObservable, observable } from 'mobx';

import { Balance } from '@/services/emigro/types';
import { getUserBalance } from '@/services/emigro/users';
import { CryptoAsset } from '@/types/assets';

export class BalanceStore {
  userBalance: Balance[] = [];
  lastUpdate: number | null = null;

  constructor() {
    makeAutoObservable(this, {
      userBalance: observable,
      setUserBalance: action,
    });
  }

  setUserBalance(balance: Balance[]): void {
    this.userBalance = balance;
    this.lastUpdate = Date.now();
  }

  find(assetCode: string): Balance | undefined {
    return this.userBalance.find((balance) => balance.assetCode === assetCode);
  }

  get(assetCode: string): number {
    const found = this.find(assetCode);
    if (found) return Number(found.balance); //TODO: change the balance to number
    return 0;
  }

  currentAssets(): CryptoAsset[] {
    return this.userBalance.map((balance) => {
      return CryptoAsset[balance.assetCode as keyof typeof CryptoAsset];
    });
  }

  async fetchUserBalance({
    interval = 10 * 1000, // 10 seconds
    force = false,
  }: {
    force?: boolean;
    interval?: number;
  } = {}): Promise<Balance[]> {
    const now = Date.now();
    if (force || this.lastUpdate === null || now - this.lastUpdate >= interval) {
      console.debug('Fetching user balance...');
      const balances = await getUserBalance();
      if (balances) {
        this.setUserBalance(balances);
      }
    }
    return this.userBalance;
  }
}

export const balanceStore = new BalanceStore();
