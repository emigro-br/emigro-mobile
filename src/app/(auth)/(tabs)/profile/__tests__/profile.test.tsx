import React from 'react';

import { fireEvent, waitFor } from '@testing-library/react-native';
import { useRouter } from 'expo-router';

import { render } from 'test-utils';

import { UserProfile } from '@/services/emigro/types';
import { balanceStore } from '@/stores/BalanceStore';
import { sessionStore } from '@/stores/SessionStore';
import { CryptoAsset } from '@/types/assets';

import Profile from '..';

jest.mock('expo-clipboard');

jest.mock('@/stores/SessionStore', () => ({
  sessionStore: {
    clear: jest.fn(),
    publicKey: 'test-public',
    get profile() {
      return null;
    },
  },
}));

const profileMock = {
  given_name: 'Test Name',
  family_name: 'Test Last Name',
  email: 'test@email.com',
  address: 'Test Address',
} as UserProfile;

describe('Profile screen', () => {
  let router: any;
  beforeEach(() => {
    jest.clearAllMocks();
    jest.useFakeTimers();
    router = useRouter();
  });

  test('Should appear loading screen while fetching the user information', () => {
    jest.spyOn(sessionStore, 'profile', 'get').mockReturnValue(null);
    const { getByTestId } = render(<Profile />);

    expect(getByTestId('loading-spinner')).toBeOnTheScreen();
  });

  test('Should render the Profile screen correctly', () => {
    jest.spyOn(sessionStore, 'profile', 'get').mockReturnValue(profileMock);
    const screen = render(<Profile />);

    expect(screen.getByRole('header')).toHaveTextContent('Test Name Test Last Name');
    expect(screen.toJSON()).toMatchSnapshot();
  });

  test('should navigate to Personal Info', () => {
    const { getByTestId } = render(<Profile />);

    const personalInfoButton = getByTestId('personal-info-button');

    fireEvent.press(personalInfoButton);

    expect(router.push).toHaveBeenCalledWith('/profile/personal-info');
  });

  test('should navigate to Configure PIN', () => {
    const { getByTestId } = render(<Profile />);

    const personalInfoButton = getByTestId('configure-pin-button');

    fireEvent.press(personalInfoButton);

    expect(router.push).toHaveBeenCalledWith('/settings/configure-pin');
  });

  test('should navigate to Bank Currency', async () => {
    jest.spyOn(balanceStore, 'currentAssets').mockReturnValue([CryptoAsset.USDC]);

    const { getByTestId, queryByTestId } = render(<Profile />);
    // expect(queryByTestId('asset-list-action-sheet"')).toBeNull();
    expect(queryByTestId('asset-avatarr"')).toBeNull();

    const personalInfoButton = getByTestId('bank-currency-button');

    fireEvent.press(personalInfoButton);

    await waitFor(() => {
      // expect(getByTestId('asset-list-action-sheet"')).toBeOnTheScreen();
      expect(getByTestId('asset-avatar')).toBeOnTheScreen(); // FIXME: could not find sset-list-action-sheet
    });
  });

  test('Should trigger the Logout action and clear the storage', () => {
    const { getByText } = render(<Profile />);

    const logoutButton = getByText('Logout');

    fireEvent.press(logoutButton);

    expect(sessionStore.clear).toHaveBeenCalled();
  });
});
