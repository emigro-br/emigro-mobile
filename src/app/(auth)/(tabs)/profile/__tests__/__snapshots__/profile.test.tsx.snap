// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile screen Should render the Profile screen correctly 1`] = `
<View
  style={
    [
      {},
      {
        "flex": 1,
        "height": "100%",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    alwaysBounceVertical={true}
    onContentSizeChange={null}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminationRequest={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onScrollShouldSetResponder={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    pagingEnabled={false}
    scrollViewRef={null}
    sendMomentumEvents={false}
    snapToEnd={true}
    snapToStart={true}
    style={
      [
        {
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "overflow": "scroll",
        },
        {
          "paddingTop": 0,
        },
      ]
    }
  >
    <RCTScrollContentView
      collapsable={false}
      style={
        [
          false,
          undefined,
        ]
      }
    >
      <View
        className="flex-1 justify-between"
      >
        <View
          className="flex-col gap-4 p-4"
        >
          <View
            className="justify-center items-center"
          >
            <View>
              <Text>
                TN
              </Text>
            </View>
            <RCTText
              accessibilityRole="header"
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              isHighlighted={false}
              selectionColor={null}
              style={
                {
                  "fontSize": 37.44,
                  "fontWeight": "bold",
                  "marginVertical": 32,
                }
              }
            >
              Test Name Test Last Name
            </RCTText>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              dataSet={
                {
                  "active": "false",
                  "disabled": "false",
                  "focus": "false",
                  "focusVisible": "false",
                  "hover": "false",
                }
              }
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              role="button"
            >
              <Text
                dataSet={
                  {
                    "active": false,
                    "disabled": undefined,
                    "focus": false,
                    "focusVisible": false,
                    "hover": false,
                  }
                }
                states={
                  {
                    "active": false,
                    "disabled": undefined,
                    "focus": false,
                    "focusVisible": false,
                    "hover": false,
                  }
                }
              >
                test-...ublic
              </Text>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="100%"
                bbWidth="100%"
                focusable={false}
                meetOrSlice={0}
                minX={0}
                minY={0}
                role="img"
                stroke="currentColor"
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": "100%",
                      "width": "100%",
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                    ]
                  }
                  stroke={
                    {
                      "type": 2,
                    }
                  }
                >
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  >
                    <RNSVGPath
                      d="M20 9H11C9.89543 9 9 9.89543 9 11V20C9 21.1046 9.89543 22 11 22H20C21.1046 22 22 21.1046 22 20V11C22 9.89543 21.1046 9 20 9Z"
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "strokeWidth",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      strokeLinecap={1}
                      strokeLinejoin={1}
                      strokeWidth="2"
                    />
                    <RNSVGPath
                      d="M5 15H4C3.46957 15 2.96086 14.7893 2.58579 14.4142C2.21071 14.0391 2 13.5304 2 13V4C2 3.46957 2.21071 2.96086 2.58579 2.58579C2.96086 2.21071 3.46957 2 4 2H13C13.5304 2 14.0391 2.21071 14.4142 2.58579C14.7893 2.96086 15 3.46957 15 4V5"
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "strokeWidth",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      strokeLinecap={1}
                      strokeLinejoin={1}
                      strokeWidth="2"
                    />
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
          <View
            className="flex-col gap-5"
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              testID="personal-info-button"
            >
              <View
                className="flex-row justify-between items-center"
              >
                <View
                  className="flex-row gap-3 items-center"
                >
                  <View
                    className="flex-col gap-1"
                  >
                    <RCTText
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      isHighlighted={false}
                      selectionColor={null}
                    >
                      Personal Info
                    </RCTText>
                  </View>
                </View>
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="100%"
                  bbWidth="100%"
                  focusable={false}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  role="img"
                  stroke="currentColor"
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "flex": 0,
                        "height": "100%",
                        "width": "100%",
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                >
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    propList={
                      [
                        "stroke",
                      ]
                    }
                    stroke={
                      {
                        "type": 2,
                      }
                    }
                  >
                    <RNSVGGroup
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                    >
                      <RNSVGPath
                        d="M9 18L15 12L9 6"
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "strokeWidth",
                            "strokeLinecap",
                            "strokeLinejoin",
                          ]
                        }
                        strokeLinecap={1}
                        strokeLinejoin={1}
                        strokeWidth="2"
                      />
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              testID="configure-pin-button"
            >
              <View
                className="flex-row justify-between items-center"
              >
                <View
                  className="flex-row gap-3 items-center"
                >
                  <View
                    className="flex-col gap-1"
                  >
                    <RCTText
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      isHighlighted={false}
                      selectionColor={null}
                    >
                      Configure your PIN
                    </RCTText>
                  </View>
                </View>
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="100%"
                  bbWidth="100%"
                  focusable={false}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  role="img"
                  stroke="currentColor"
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "flex": 0,
                        "height": "100%",
                        "width": "100%",
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                >
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    propList={
                      [
                        "stroke",
                      ]
                    }
                    stroke={
                      {
                        "type": 2,
                      }
                    }
                  >
                    <RNSVGGroup
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                    >
                      <RNSVGPath
                        d="M9 18L15 12L9 6"
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "strokeWidth",
                            "strokeLinecap",
                            "strokeLinejoin",
                          ]
                        }
                        strokeLinecap={1}
                        strokeLinejoin={1}
                        strokeWidth="2"
                      />
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              testID="bank-currency-button"
            >
              <View
                className="flex-row justify-between items-center"
              >
                <View
                  className="flex-row gap-3 items-center"
                >
                  <View
                    className="flex-col gap-1"
                  >
                    <RCTText
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      isHighlighted={false}
                      selectionColor={null}
                    >
                      Bank account currency: not set
                    </RCTText>
                    <RCTText
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      isHighlighted={false}
                      selectionColor={null}
                    >
                      Used for deposit and withdraw
                    </RCTText>
                  </View>
                </View>
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="100%"
                  bbWidth="100%"
                  focusable={false}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  role="img"
                  stroke="currentColor"
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "flex": 0,
                        "height": "100%",
                        "width": "100%",
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                >
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    propList={
                      [
                        "stroke",
                      ]
                    }
                    stroke={
                      {
                        "type": 2,
                      }
                    }
                  >
                    <RNSVGGroup
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                    >
                      <RNSVGPath
                        d="M9 18L15 12L9 6"
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "strokeWidth",
                            "strokeLinecap",
                            "strokeLinejoin",
                          ]
                        }
                        strokeLinecap={1}
                        strokeLinejoin={1}
                        strokeWidth="2"
                      />
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              dataSet={
                {
                  "active": "false",
                  "disabled": "false",
                  "focus": "false",
                  "focusVisible": "false",
                  "hover": "false",
                }
              }
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              role="button"
            >
              <Text
                dataSet={
                  {
                    "active": false,
                    "disabled": undefined,
                    "focus": false,
                    "focusVisible": false,
                    "hover": false,
                  }
                }
                states={
                  {
                    "active": false,
                    "disabled": undefined,
                    "focus": false,
                    "focusVisible": false,
                    "hover": false,
                  }
                }
              >
                Delete account
              </Text>
            </View>
          </View>
        </View>
        <View
          className="items-center mb-12"
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            dataSet={
              {
                "active": "false",
                "disabled": "false",
                "focus": "false",
                "focusVisible": "false",
                "hover": "false",
              }
            }
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            role="button"
          >
            <Text
              dataSet={
                {
                  "active": false,
                  "disabled": undefined,
                  "focus": false,
                  "focusVisible": false,
                  "hover": false,
                }
              }
              states={
                {
                  "active": false,
                  "disabled": undefined,
                  "focus": false,
                  "focusVisible": false,
                  "hover": false,
                }
              }
            >
              Logout
            </Text>
          </View>
          <RCTText
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            isHighlighted={false}
            selectionColor={null}
          >
            version 
            mock
             (
            mock
            )
          </RCTText>
        </View>
      </View>
    </RCTScrollContentView>
  </RCTScrollView>
</View>
`;
